x-pack/plugins/uptime/public/state/reducers/synthetics.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createAction, handleActions, Action } from 'redux-actions';
+import {
+  isScreenshotBlockDoc,
+  ScreenshotBlockDoc,
+} from '../../../common/runtime_types/ping/synthetics';
+
+export interface PendingBlock {
+  status: 'pending' | 'loading';
+}
+
+export function isPendingBlock(data: unknown): data is PendingBlock {
```

Comment:
I think it makes sense to keep it here, since it's alongside the type it checks for. You can make the argument that all these types/interfaces should be extracted to a separate file. But then do we have a 1:1 between types/reducers? Or one huge types file? We also currently have types living in actions files in other places.

I think Shahzad and I have found over time it's helpful to keep the actions/types/reducer all together, and have them divided up by domain rather than function.